### PR TITLE
KEYCLOAK-16679 Add algorithm settings for client assertion signature in OIDC identity broker

### PR DESCRIPTION
**Pull request by i7a7467**
https://issues.redhat.com/browse/KEYCLOAK-16679

### CODE REVIEW

File: themes/src/main/resources/theme/base/admin/messages/admin-messages_en.properties [hunk 1]
```diff
@@ -644,6 +644,8 @@ client-auth.client_secret_jwt=Client secret as jwt
 client-auth.private_key_jwt=JWT signed with private key
 identity-provider.client-id.tooltip=The client or client identifier registered within the identity provider.
 client-secret=Client Secret
+client-assertion-signing-algorithm=Client Assertion Signature Algorithm
+client-assertion-signing-algorithm.tooltip=JWA Algorithm used for client assertions.
```
Comment (mposolda) [inline themes/src/main/resources/theme/base/admin/messages/admin-messages_en.properties:648]: Is it possible to add slightly more details to this tooltip to specify what exactly does this option mean, when it is needed to be configured etc? My vote is just include everything, which you added to the documentation to this tooltip. So something like this: `Signature algorithm to create JWT assertion as client authentication. In the case of JWT signed with private key or Client secret as jwt, it is required. If no algorithm is specified, the following algorithm is adapted. `RS256` is adapted the in the case of JWT signed with private key.  `HS256` is adapted the in the case of Client secret as jwt.`

**Review by mposolda — CHANGES_REQUESTED** [review]
@i7a7467 Thanks for the PR and sorry for the late review. Few comments:
- Rebase will be needed
- I suggested inline to slightly change the tooltip to make it more clear what does this means
- Automated test will be needed

**Comment by mposolda** [comment]
Docs PR: https://github.com/keycloak/keycloak-documentation/pull/1075

**Comment by i7a7467** [comment]
@mposolda 
Thank you for your review comment.
I fixed the review points. I rebased.

### About automated tests
I added tests using private_key_jwt and client_secret_jwt authentication methods and non-default algorithms.
- private_key_jwt:ES256
- client_secret_jwt:HS384

File: themes/src/main/resources/theme/base/admin/messages/admin-messages_en.properties (hunk 1 above)
Comment (i7a7467) [reply themes/src/main/resources/theme/base/admin/messages/admin-messages_en.properties]: That's exactly what you say. 
I added the same content as the documentation to tooltip message.

**Review by mposolda — APPROVED** [review]
@i7a7467 Thanks!

**Pull request merged**